tests/test_spiderloader/__init__.py

```diff
@@ -89,3 +90,14 @@ def test_crawler_runner_loading(self):
         crawler = runner.create_crawler('spider1')
         self.assertTrue(issubclass(crawler.spidercls, scrapy.Spider))
         self.assertEqual(crawler.spidercls.name, 'spider1')
+
+    def test_bad_spider_modules_warning(self):
+
+        with warnings.catch_warnings(record=True) as w:
+            module = 'tests.test_spiderloader.test_spiders.doesnotexist'
+            settings = Settings({'SPIDER_MODULES': [module]})
+            spider_loader = SpiderLoader.from_settings(settings)
+            self.assertIn("Could not load spiders from module", str(w[0].message))
```

Comment:
Yeah, I think this information should be available somehow, because otherwise it could make it harder to debug than before.

You also said that the root cause is that scrapy loads everything before running commands like `scrapy version`; it makes sense because users can add their own commands and you need to load them all before executing a command. The approach in this PR looks good to me - if discovery fails it is fine to still allow running built-in commands.